### PR TITLE
Move to golang 1.16.7

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Update golang to 1.16.7

###### Change Log  

- golang spec and all spec which BuildRequires golang

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2020-24455
- https://nvd.nist.gov/vuln/detail/CVE-2021-29923

###### Does this affect the toolchain?  

**YES**

### CODE REVIEW

**Review by anphel31 — COMMENTED** [review]
Can you please update the golang version in our workflow files as well?
.github/workflows/go-test-coverage.yml
and
.github/workflows/quickstart_1.0.yml 

[nicolas] => done

**Review by jslobodzian — APPROVED** [review]

File: cgmanifest.json [hunk 1]
```diff
@@ -4135,7 +4135,7 @@
         "type": "other",
         "other": {
           "name": "moby-runc",
-          "version": "1.0.0-rc95+azure",
+          "version": "1.0.0~rc95+azure",
```
[the remark this answers is not in the document]
Comment (nicogbg) [reply cgmanifest.json:4138]: fixed

File: toolkit/docs/building/prerequisites.md [hunk 2]
```diff
@@ -11,13 +11,13 @@ sudo add-apt-repository ppa:longsleep/golang-backports
 sudo apt-get update
 
 # Install required dependencies.
-sudo apt -y install make tar wget curl rpm qemu-utils golang-1.15-go genisoimage python-minimal bison gawk parted
+sudo apt -y install make tar wget curl rpm qemu-utils golang-1.16-go genisoimage python-minimal bison gawk parted
 
 # Recommended but not required: `pigz` for faster compression operations.
 sudo apt -y install pigz
 
 # Fix go 1.15 link
```
[the remark this answers is not in the document]
Comment (nicogbg) [reply toolkit/docs/building/prerequisites.md:19]: ?

File: toolkit/scripts/toolchain/build_official_toolchain_rpms.sh [hunk 3]
```diff
@@ -492,7 +492,7 @@ chroot_and_install_rpms gperf
 chroot_and_install_rpms cryptsetup
 build_rpm_in_chroot_no_install systemd
 
-build_rpm_in_chroot_no_install golang-1.15
+build_rpm_in_chroot_no_install golang-1.16
```
[the remark this answers is not in the document]
Comment (nicogbg) [reply toolkit/scripts/toolchain/build_official_toolchain_rpms.sh:495]: yep, this guy hit me hard :( Andrew just warned me about it yesterday...
This is now fixed

File: SPECS/flannel/flannel.spec [hunk 4]
```diff
@@ -58,6 +58,8 @@ rm -rf %{gopath_flannel_folder}/*
 %{_bindir}/flanneld
 
 %changelog
+* Fri Aug 06 2021 Nicolas Guibourge <nicolasg@microsoft.com> 0.12.0-3
```
[the remark this answers is not in the document]
Comment (nicogbg) [reply SPECS/flannel/flannel.spec:61]: thanks

File: toolkit/docs/building/prerequisites.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicogbg) [reply toolkit/docs/building/prerequisites.md]: fixed

**Review by PawelWMS — APPROVED** [review]

File: cgmanifest.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (anphel31) [reply cgmanifest.json:4138]: looks like this change is still not reverted to dash - unresolving as reminder to double check

[the remark this answers is not in the document]
Comment (nicogbg) [reply cgmanifest.json:4198]: done (I hope this time)

**Review by jslobodzian — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

File: cgmanifest.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicogbg) [reply cgmanifest.json]: this is done but fail cgmanisfest.json automated verif. Is that on purpose ?

**Comment by mfrw** [comment]
While we are here, can we also take care of : https://nvd.nist.gov/vuln/detail/CVE-2021-29923

**Comment by nicogbg** [comment]
force merge because moby-runc must stay as it is in cgmanifest.json

**Pull request merged**